### PR TITLE
Expand hardware info retrieval and add read_board to debug_snapshot

### PR DESCRIPTION
**Pull request by mehwang**
@ablasdel, @rctoris:
This expands the regular debug_snapshot command with hardware info using suggestions from @dbking77 and @alexhenning.

### CODE REVIEW

**Comment by rctoris** [comment]
:+1: LGTM

**Comment by ablasdel** [comment]
:+1: :eyes:

File: src/fetch_tools/commands/debug_snapshot.py [hunk 1]
```diff
@@ -35,8 +35,8 @@
 # All commands requiring sudo must be added in main
 commands = {"dpkg_fetch":"COLUMNS=200 dpkg -l ros-indigo-fetch-*",
             "dpkg_all":"COLUMNS=200 dpkg -l",
```
Comment (erelson) [inline src/fetch_tools/commands/debug_snapshot.py]: Saw the COLUMNS trick here for the first time... Question: was COLUMNS=200 set to reduce the size of the logs? 

(If it was set to make sure the output was wide enough, I'm getting a width of >400 by default when using dpkg via e.g. subprocess.check_output, so it's probably wide enough to begin.)

Comment (ablasdel) [inline src/fetch_tools/commands/debug_snapshot.py]: Yeah when you don't have an actual terminal the column width gets messed up on dpkg. You need that to ensure you can actually read it in the text file that we generate.

**Pull request merged**